### DISCUSSION TITLE
Prometheus remote write unable to read

### DISCUSSION THREAD

**Discussion by prateek1192** [vectordotdev/vector#15114] [Q&A]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Hello, I am trying to send data to vector via prometheus remote write so that I can further send it to datadog. I have deployed a vector helm chart in the same namespace as prometheus is running. I see the vector service and the headless service deployed in the namespace and I am trying to make prometheus send data to vector. I see the following log messages in prometheus

`ts=2022-11-01T00:33:27.942Z caller=dedupe.go:112 component=remote level=warn remote_name=fa13dd url=http://vector.istio-system.svc.cluster.local:9090/ msg="Failed to send batch, retrying" err="Post \"http://vector.istio-system.svc.cluster.local:9090/\": read tcp 192.168.135.32:46236->10.100.5.151:9090: read: connection reset by peer"
ts=2022-11-01T00:33:31.632Z caller=dedupe.go:112 component=remote level=warn remote_name=fa13dd url=http://vector.istio-system.svc.cluster.local:9090/ msg="Failed to send batch, retrying" err="Post \"http://vector.istio-system.svc.cluster.local:9090/\": read tcp 192.168.135.32:50902->10.100.5.151:9090: read: connection reset by peer"
ts=2022-11-01T00:33:32.949Z caller=dedupe.go:112 component=remote level=warn remote_name=fa13dd url=http://vector.istio-system.svc.cluster.local:9090/ msg="Failed to send batch, retrying" err="Post \"http://vector.istio-system.svc.cluster.local:9090/\": read tcp 192.168.135.32:50904->10.100.5.151:9090: read: connection reset by peer"`

I can see that the vector instance starts with the following message so the remote write source is running. I am running vector as an aggregator using the default helm install command helm install vector vector/vector --namespace istio-system. I see that the documentation states that we need to run vector in daemon mode for prometheus remote write but is there a way that this use case is supported ?

`kubectl logs vector-0 -n istio-system
2022-10-31T23:25:37.008286Z  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info,buffers=info,kube=info"
2022-10-31T23:25:37.008359Z  INFO vector::app: Loading configs. paths=["/etc/vector"]
2022-10-31T23:25:37.009904Z  INFO vector::topology::running: Running healthchecks.
2022-10-31T23:25:37.010087Z  INFO vector: Vector has started. debug="false" version="0.24.1" arch="x86_64" build_id="8935681 2022-09-12"
2022-10-31T23:25:37.010114Z  INFO vector::internal_events::api: API server running. address=127.0.0.1:8686 playground=off
2022-10-31T23:25:37.010229Z  INFO vector::topology::builder: Healthcheck: Passed.
2022-10-31T23:25:37.012883Z  INFO source{component_kind="source" component_id=prom_remote_write component_type=prometheus_remote_write component_name=prom_remote_write}: vector::sources::util::http::prelude: Building HTTP server. address=0.0.0.0:9090`

### Configuration

```text
Following is the config file I added for remote write in prometheus config map

    remote_write:
      - url: http://vector.istio-system.svc.cluster.local:9090

Following is the config for my vector instance for the remote write source

{
  "sources": {
    "my_source_id": {
      "type": "prometheus_remote_write",
      "acknowledgements": null,
      "address": "0.0.0.0:9090"
    }
  }
}
```

### Version

vector 0.24.1 (x86_64-unknown-linux-gnu 8935681 2022-09-12)

**Comment by jszwedko** [comment]
(moved to a discussion since this seems like more of a question than a bug report)

Hi @prateek1192 !

I'd suggest a couple of things:

* Running Vector with debug logs (`VECTOR_LOG=debug`)
* Trying to run the code publishing via remote write and Vector outside of Kubernetes to see if you see the same connection reset. This would help narrow down if it is Vector closing the connections or if you have another network connectivity issue between the containers.

**Reply by prateek1192** [reply]
I tried running with vector with debug. I saw one more line showing the server for Prometheus remote write to be started. After that while trying to debug, I moved the vector helm chart from istio-system namespace to another namespace and I was able to see the metrics in vector. Is this a known thing ?

**Reply by jszwedko** [reply]
This sounds like an issue with network connectivity outside the scope of Vector 🤔

**Reply by prateek1192** [reply]
I actually had enabled internal_metrics with stdout and when I saw data in vector logs , I assumed it was from prometheus but it was only from internal metrics. I would update when I am able to find the reason for this happening. Meanwhile , I have noticed that in the docs https://vector.dev/docs/reference/configuration/sinks/datadog_metrics/ if I use null for acknowledgements and healthcheck, I get an error saying `2022-11-10T01:32:25.181666Z ERROR vector::cli: Configuration error. error=sinks.ddog.acknowledgements: invalid type: unit value, expected any valid TOML value at line 17 column 23` . I am using YAML. I switch to acknowledgements: false , it works. Is it a known issue or did I do something wrong ?

**Reply by jszwedko** [reply]
Hmm, can you share the full config you were using when you got that error?

**Reply by prateek1192** [reply]
`prateek:charts prateek.arora$ kubectl describe cm vector -n cluster
Name:         vector
Namespace:    cluster
Labels:       app.kubernetes.io/component=Aggregator
              app.kubernetes.io/instance=vector
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=vector
              app.kubernetes.io/version=0.24.1-distroless-libc
              helm.sh/chart=vector-0.16.3
Annotations:  meta.helm.sh/release-name: vector
              meta.helm.sh/release-namespace: cluster

Data

aggregator.yaml:

data_dir: /vector-data-dir
api:
  enabled: true
  address: 127.0.0.1:8686
  playground: false
sources:
  prom_remote_write:
    address: 0.0.0.0:6000
    type: vector
    version: "2"
sinks:
  datadog_ext:
    type: datadog_metrics
    inputs: 
      - prom_remote_write
    default_api_key: "59exxxxxxxxxxxxxxxxxxxxxxa"
    acknowledgements: null
    healthcheck: null
  stdout:
    type: console
    inputs: [prom_remote_write]
    encoding:
      codec: json


BinaryData


Events:  <none>
`

and here are the logs

`kubectl logs vector-0 -n cluster
2022-11-10T21:59:10.954102Z  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info,buffers=info,kube=info"
2022-11-10T21:59:10.954173Z  INFO vector::app: Loading configs. paths=["/etc/vector"]
2022-11-10T21:59:10.954668Z ERROR vector::cli: Configuration error. error=sinks.datadog_ext.acknowledgements: invalid type: unit value, expected any valid TOML value at line 17 column 23`

**Reply by jszwedko** [reply]
I'd try deleting the `acknowledgements: null` and `healthcheck: null` lines.

**Reply by prateek1192** [reply]
Thanks for the suggestion. I followed the official docs which had these as null https://vector.dev/docs/reference/configuration/sources/prometheus_remote_write/ and as such I used null.
I have used acknowledgements: false and healthcheck: false and that worked. I couldn't get vector to work when istio-proxy injection is set to true. As such I am trying to use other solutions now. Thanks for the help, hope I can use vector with istio some day.
